File: src/main/java/net/evenh/coffeebot/machine/MachineRepository.java
package net.evenh.coffeebot.machine;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface MachineRepository extends JpaRepository<Machine, Long> {
  Optional<Machine> findOneByNameIgnoringCase(String name);
}

File: src/main/java/net/evenh/coffeebot/machine/MachineRepository.kt
package net.evenh.coffeebot.machine

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface MachineRepository : JpaRepository<Machine, Long> {
    fun findOneByNameIgnoringCase(name: String): Optional<Machine>
}
